src/Compilers/CSharp/Portable/Binder/Binder_Lambda.cs

```diff
@@ -237,6 +243,26 @@ static void checkAttributes(AnonymousFunctionExpressionSyntax syntax, SyntaxList
             }
         }
 
+        private (RefKind, TypeWithAnnotations) BindExplicitLambdaReturnType(TypeSyntax syntax, BindingDiagnosticBag diagnostics)
+        {
+            MessageID.IDS_FeatureLambdaReturnType.CheckFeatureAvailability(diagnostics, syntax);
+
+            syntax = syntax.SkipRef(out RefKind refKind);
+            var returnType = BindType(syntax, diagnostics);
+            var type = returnType.Type;
+
+            if (returnType.IsStatic)
+            {
+                diagnostics.Add(ErrorFacts.GetStaticClassReturnCode(useWarning: false), syntax.Location, type);
+            }
+            else if (returnType.IsRestrictedType(ignoreSpanLikeTypes: true))
```

Comment:
Theoretically, when we start inferring delegate types, we could encounter an async function with an `__arglist` parameter. Do we need to check that here? Or perhaps assert that we'll never encounter such a scenario? SourceOrdinaryMethodSymbolBase checks this case in its equivalent method, so I think we should do something about it, even if it's just to assert we'll never hit it.

Reply:
Is this an issue introduced by explicit return types or an existing issue with lambda parameters?

It looks like the parser rejects `__arglist` in `IsPossibleLambdaParameter()`. I've added some parsing tests.